Index: Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Gardener.java
===================================================================
diff --git a/Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Gardener.java b/Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Gardener.java
deleted file mode 100644
--- a/Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Gardener.java	(revision 44689cca1952ec452c7bd1d32aee5c8150756da9)
+++ /dev/null	(revision 44689cca1952ec452c7bd1d32aee5c8150756da9)
@@ -1,37 +0,0 @@
-package ejercicio1.hilos;
-
-import lombok.Data;
-
-@Data
-public class Gardener implements Runnable {
-    private int id;
-    private int contador = 0;
-    final Town town = new Town();
-
-    public Gardener(int id) {
-        this.id = id;
-    }
-
-    @Override
-    public void run() {
-    for (int i = 0; i < 10; i++) {
-        synchronized (town) {
-            if (!town.isGardenFree(i)) {
-                town.getGardens().set(i, true);
-                contador++;
-                System.out.println("Gardener " + id + " is working in garden " + i);
-                try {
-                    Thread.sleep(1000L);
-                } catch (InterruptedException e) {
-                    e.printStackTrace();
-                }
-                System.out.println("Gardener " + id + " has finished working in garden " + i);
-                //town.getGardens().set(i, false);
-            } else {
-                System.out.println("Gardener " + id + " is waiting");
-            }
-        }
-
-    }
-}
-}
Index: Tema-1/Pruebas/Prueba_Examen/src/main/java/Main.java
===================================================================
diff --git a/Tema-1/Pruebas/Prueba_Examen/src/main/java/Main.java b/Tema-1/Pruebas/Prueba_Examen/src/main/java/Main.java
deleted file mode 100644
--- a/Tema-1/Pruebas/Prueba_Examen/src/main/java/Main.java	(revision 44689cca1952ec452c7bd1d32aee5c8150756da9)
+++ /dev/null	(revision 44689cca1952ec452c7bd1d32aee5c8150756da9)
@@ -1,14 +0,0 @@
-import ejercicio1.hilos.Gardener;
-import ejercicio1.hilos.Town;
-
-public class Main {
-    public static void main(String[] args) {
-
-        for (int i = 0; i <= 2; i++) {
-            Gardener gardener = new Gardener(i);
-            Thread thread = new Thread(gardener);
-            thread.start();
-            s
-        }
-    }
-}
Index: Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Town.java
===================================================================
diff --git a/Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Town.java b/Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Town.java
deleted file mode 100644
--- a/Tema-1/Pruebas/Prueba_Examen/src/main/java/ejercicio1/hilos/Town.java	(revision 44689cca1952ec452c7bd1d32aee5c8150756da9)
+++ /dev/null	(revision 44689cca1952ec452c7bd1d32aee5c8150756da9)
@@ -1,26 +0,0 @@
-package ejercicio1.hilos;
-
-import lombok.Data;
-
-import java.util.ArrayList;
-import java.util.List;
-
-@Data
-public class Town{
-    private List<Boolean> gardens = new ArrayList<>() {{
-        add(false);
-        add(false);
-        add(false);
-        add(false);
-        add(false);
-        add(false);
-        add(false);
-        add(false);
-        add(false);
-        add(false);
-    }};
-
-    public synchronized boolean isGardenFree(int i) {
-        return gardens.get(i);
-    }
-}
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<project version=\"4\">\n  <component name=\"VcsDirectoryMappings\">\n    <mapping directory=\"\" vcs=\"Git\" />\n  </component>\n</project>
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/vcs.xml b/.idea/vcs.xml
--- a/.idea/vcs.xml	(revision 44689cca1952ec452c7bd1d32aee5c8150756da9)
+++ b/.idea/vcs.xml	(date 1701107071073)
@@ -2,5 +2,6 @@
 <project version="4">
   <component name="VcsDirectoryMappings">
     <mapping directory="" vcs="Git" />
+    <mapping directory="$PROJECT_DIR$/Tema-2/Ejercicio1" vcs="Git" />
   </component>
 </project>
\ No newline at end of file
Index: Tema-1/Pruebas/Prueba_Examen/src/main/java/GestorDescargas.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Tema-1/Pruebas/Prueba_Examen/src/main/java/GestorDescargas.java b/Tema-1/Pruebas/Prueba_Examen/src/main/java/GestorDescargas.java
new file mode 100644
--- /dev/null	(date 1700158877950)
+++ b/Tema-1/Pruebas/Prueba_Examen/src/main/java/GestorDescargas.java	(date 1700158877950)
@@ -0,0 +1,34 @@
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class GestorDescargas {
+    public static void main(String[] argumentos) throws IOException {
+        GestorDescargas gd = new GestorDescargas();
+        gd.descargarArchivo("https://www.boe.es/legislacion/documentos/ConstitucionCASTELLANO.pdf", "resultado.txt");
+        BufferedReader br = new BufferedReader(new FileReader("resultado.txt"));
+        String linea;
+        while ((linea = br.readLine()) != null) {
+            System.out.println(linea);
+        }
+    }
+
+    public void descargarArchivo(String url_descargar, String nombreArchivo) {
+        System.out.println("Descargando " + url_descargar);
+        try {
+            URL laUrl = new URL(url_descargar);
+            BufferedReader bReader = new BufferedReader(new InputStreamReader(laUrl.openStream()));
+            FileWriter escritorFichero = new FileWriter(nombreArchivo);
+            String linea;
+            while ((linea = bReader.readLine()) != null) {
+                escritorFichero.write(linea);
+            }
+            escritorFichero.close();
+            bReader.close();
+        } catch (MalformedURLException e) {
+            System.out.println("URL mal escrita!");
+        } catch (IOException e) {
+            System.out.println("Fallo en la lectura del fichero");
+        }
+    }
+}
\ No newline at end of file
